src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -54,7 +54,7 @@ type mgmtModule struct {
 }
 
 func (mod *mgmtModule) HandleCall(session *drpc.Session, method drpc.Method, req []byte) ([]byte, error) {
-	if method != drpc.MethodGetAttachInfo {
+	if method != drpc.MethodGetAttachInfo && method != drpc.MethodDisconnect {
 		return nil, drpc.UnknownMethodFailure()
 	}
```

Comment:
I think Joel's original intent of this check here was to prevent the work being done if we get a bad method up top. If we dont want to short circuit the call here then I should add a default case in the switch below.

Reply:
I think this made sense as an optimization when there was only one method possibility. If we want to avoid doing any work, we could break the logic to get the PID into a helper function, and call it in the respective methods, instead of doing the work before the switch.